src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/RazorFormattingEndpoint.cs

```diff
@@ -29,20 +26,18 @@ internal class RazorFormattingEndpoint : IDocumentFormattingHandler, IDocumentRa
         private readonly DocumentResolver _documentResolver;
         private readonly RazorFormattingService _razorFormattingService;
         private readonly RazorDocumentMappingService _razorDocumentMappingService;
-        private readonly AdhocWorkspaceFactory _adhocWorkspaceFactory;
         private readonly IOptionsMonitor<RazorLSPOptions> _optionsMonitor;
         private readonly ILogger _logger;
 
         private static readonly IReadOnlyList<string> s_csharpTriggerCharacters = new[] { "}", ";" };
-        private static readonly IReadOnlyList<string> s_htmlTriggerCharacters = Array.Empty<string>();
+        private static readonly IReadOnlyList<string> s_htmlTriggerCharacters = new[] { "\n", "{", "}", ";" };
```

Comment:
`\n` totally doesn't work as an on type formatting trigger, we just never get a message. I'm guessing platform is ignore it and doing OnAutoInsert instead.